.travis.yml

```diff
@@ -10,26 +10,46 @@ addons:
 
 matrix:
   include:
-    - jdk: oraclejdk9
-      env: IDE_ID=IU-2018.2
-    - jdk: oraclejdk9
-      env: IDE_ID=IU-2018.1
     - jdk: oraclejdk8
-      env: IDE_ID=IU-2017.3
+      env: IDEA_VERSION="IU-2018.2.2" PHP_PLUGIN_VERSION="182.4129.45" PSI_VIEWER_PLUGIN_VERSION="182.2757.2" SINCE_BUILD="182" UNTIL_BUILD="182.*"
+
     - jdk: oraclejdk8
-      env: IDE_ID=IU-2017.2
+      env: IDEA_VERSION="IU-2018.1.5" PHP_PLUGIN_VERSION="181.5281.35" PSI_VIEWER_PLUGIN_VERSION="2018.1.2" SINCE_BUILD="181" UNTIL_BUILD="181.*"
+
     - jdk: oraclejdk8
-      env: IDE_ID=IU-2017.1
+      env: IDEA_VERSION="IU-2017.3.5" PHP_PLUGIN_VERSION="173.4674.46" PSI_VIEWER_PLUGIN_VERSION="3.28.93" SINCE_BUILD="173" UNTIL_BUILD="173.*"
+
     - jdk: oraclejdk8
-      env: IDE_ID=IU-2016.3
+      env: IDEA_VERSION="IU-2017.2.7" PHP_PLUGIN_VERSION="172.4155.41" PSI_VIEWER_PLUGIN_VERSION="3.28.93" SINCE_BUILD="172" UNTIL_BUILD="172.*"
+
     - jdk: oraclejdk8
-      env: IDE_ID=IU-2016.2
+      env: IDEA_VERSION="IU-2017.1.6" PHP_PLUGIN_VERSION="171.4694.2" PSI_VIEWER_PLUGIN_VERSION="3.28.93" SINCE_BUILD="171" UNTIL_BUILD="171.*"
 
-script: ".travis/build.sh"
+    - jdk: oraclejdk8
+      env: IDEA_VERSION="IU-2016.3.6" PHP_PLUGIN_VERSION="163.10504.2" PSI_VIEWER_PLUGIN_VERSION="3.28.93" SINCE_BUILD="163" UNTIL_BUILD="163.*"
+
+    - jdk: oraclejdk8
+      env: IDEA_VERSION="IU-2016.2.5" PHP_PLUGIN_VERSION="162.1628.23" PSI_VIEWER_PLUGIN_VERSION="3.28.93" SINCE_BUILD="162" UNTIL_BUILD="162.*"
+
+before_cache:
+- rm -f  $HOME/.gradle/caches/modules-2/modules-2.lock
+- rm -fr $HOME/.gradle/caches/*/plugin-resolution/
+- rm -fr $HOME/.gradle/caches/*/fileHashes/fileHashes.bin
+- rm -fr $HOME/.gradle/caches/*/fileHashes/fileHashes.lock
 
 cache:
   directories:
-    - .travis/.cache
+  - $HOME/.gradle/caches/
+  - $HOME/.gradle/wrapper/
+
+before_install:
+- "export ORG_GRADLE_PROJECT_ideaVersion=${IDEA_VERSION}"
+- "export ORG_GRADLE_PROJECT_phpPluginVersion=${PHP_PLUGIN_VERSION}"
+- "export ORG_GRADLE_PROJECT_psiViewerPluginVersion=${PSI_VIEWER_PLUGIN_VERSION}"
```

Comment:
You can skip this line. All versions of PSI_Plugin are the same. So you do not need to specify (IMHO)

Reply:
We certainly could - somehow. I'm not aware on how to cascade dependencies for different environments.

If you're asking me - I'd just keep it since it's small, convenient and not embedded in the final artifacts.

Reply:
Yes, I would keep it. Assembling the array of dependencies out of band makes it just harder to read and nothing is gained from it. - IMHO.

Reply:
Let's keep it, I have to install it all the time locally.